agent/dockerclient/timeout.go

```diff
@@ -38,6 +38,9 @@ const (
 	ListContainersTimeout = 10 * time.Minute
 	// InspectContainerTimeout is the timeout for the InspectContainer API.
 	InspectContainerTimeout = 30 * time.Second
+	// TopContainerTimeout is the timeout for the TopContainer API.
+	TopContainerTimeout = 10 * time.Second
```

Comment:
how did we come up with this value?

Reply:
To be honest, I made an assumption based on https://godoc.org/docker.io/go-docker#Client.ContainerTop being similar to https://godoc.org/docker.io/go-docker#Client.ContainerInspect.  The 30 second timeout seems extensive for InspectContainerTimeout so for TopContainerTimeout I set it to 10 seconds.

Would you prefer to see profiling of the TopContainer command?
Otherwise we can change this if we find a reason.

Reply:
Some kind of profiling is usually preferable for these timeouts. 
10 seconds seem too low to me, so looking to avoid surprises.
 timeouts usually some x times more than average api call latencies.

Reply:
I'll update with profiling info

Reply:
I added some simple print statements to the integ tests and ran them to capture the timing of the `docker top` api call:
```
[Info] ###TEST### start: 2020-09-04 00:15:15.162484886 +0000 UTC m=+0.764416624
[Info] ###TEST### topContainerOut: &{[[root 5181 5119 0 00:15 ? 00:00:00 sleep 5]] [UID PID PPID C STIME TTY TIME CMD]}
topErr: <nil>
[Info] ###TEST### end: 2020-09-04 00:15:15.180661484 +0000 UTC m=+0.782593205
[Info] ###TEST### elapsed: 18.176581ms
```
```
[Info] ###TEST### start: 2020-09-04 00:15:15.222661243 +0000 UTC m=+0.824592981
[Info] ###TEST### topContainerOut: &{[[root 5263 5199 0 00:15 ? 00:00:00 sleep 5]] [UID PID PPID C STIME TTY TIME CMD]}
topErr: <nil>
[Info] ###TEST### end: 2020-09-04 00:15:15.234055193 +0000 UTC m=+0.835986926
[Info] ###TEST### elapsed: 11.393945ms
```
```
[Info] ###TEST### start: 2020-09-04 00:15:15.268344148 +0000 UTC m=+0.870275896
[Info] ###TEST### topContainerOut: &{[[root 5234 5154 0 00:15 ? 00:00:00 sleep 5]] [UID PID PPID C STIME TTY TIME CMD]}
topErr: <nil>
[Info] ###TEST### end: 2020-09-04 00:15:15.278161029 +0000 UTC m=+0.880092750
[Info] ###TEST### elapsed: 9.816854ms
```
`[Info] ###TEST### elapsed: 12.350418ms`
`[Info] ###TEST### elapsed: 13.807724ms`
`[Info] ###TEST### elapsed: 10.075021ms`
`[Info] ###TEST### elapsed: 11.396256ms`
`[Info] ###TEST### elapsed: 11.115756ms`
...
Over 100 executions my max was ~19ms and min was ~10ms for each `TopContainer` call.

Is there a specific context you're worried about this timing out in 10 seconds?

Reply:
yes, when there are multiple tasks/containers running simultaneously in the instance, docker could be stressed(inspect/start/stop containers etc), and 10s timeout could be hit faster. 
when the above profiling done was done, was docker stressed enough with other operations?

Reply:
I wrote a quick bash script to run a bunch of docker api calls to top then ran it in a bunch of tmux sessions on the instance where I also ran the top test:
```
#!/bin/bash
# this calls docker top repeatedly for a specific container
# arg 0 should be some container id from `docker ps`
# arg 1 should be a short enough time interval to cause docker pain
args=("$@")
while true; do
  docker top ${args[0]}
  sleep ${args[1]}
done
```
use: `./docker_stressor.sh <containerid> <interval>`
and engine integ tests run using `go test -tags integration -timeout=5m -v ./agent/engine/... > output`
running integ tests side-by-side with multiple tmux sessions (aka "stressors") having following configs and output:

with 2 stressors running docker top at .5 second interval:
max  `[Info] ###TEST### elapsed: 28.333578ms`
with 10 stressors running `docker top` at a .05 second interval:
max `[Info] ###TEST### elapsed: 52.632767ms`
with 10 stressors running docker top at .01 second interval:
max `[Info] ###TEST### elapsed: 89.34588ms`
with 10 stressors running docker top at .001 second interval:
max `[Info] ###TEST### elapsed: 91.587552ms`
with 10 stressors running docker top at .000001 second interval
max `[Info] ###TEST### elapsed: 86.837214ms`

I will note that there might be a limit to the `sleep` interval that I'm using -- I can't find anything in the man page (which points to `info coreutils 'sleep invocation'`)

Despite the above, I'll set the timeout to 30 seconds to be in line with the inspect container timeout to be extra-conservative.

Reply:
lgtm. thanks for testing